### PR TITLE
[FIX] Default server language not being applied

### PR DESCRIPTION
**Pull request by sampaiodiego**
With the new logic, the user's language takes precedence, then the language stored on `localStorage`, then default server's language.

### CODE REVIEW

File: client/startup/i18n.js [hunk 1]
```diff
@@ -56,29 +59,17 @@ Meteor.startup(() => {
...
-		const defaultLanguage = userLanguage || RocketChat.settings.get('Language') || 'en';
-
-		if (!currentLanguage.get()) {
-			setLanguage(defaultLanguage);
-		}
-
-		if (userLanguage && userLanguage !== currentLanguage.get()) {
-			setLanguage(userLanguage);
-		}
+		const user = RocketChat.models.Users.findOne(Meteor.userId(), { fields: { language: 1 }});
```
Comment (tassoevan) [inline client/startup/i18n.js]: I've to read this line five times before get it... :bug:

File: client/startup/i18n.js [hunk 2]
```diff
@@ -56,29 +59,17 @@ Meteor.startup(() => {
...
-
-		const defaultLanguage = userLanguage || RocketChat.settings.get('Language') || 'en';
-
-		if (!currentLanguage.get()) {
-			setLanguage(defaultLanguage);
-		}
-
-		if (userLanguage && userLanguage !== currentLanguage.get()) {
-			setLanguage(userLanguage);
-		}
```
Comment (tassoevan) [inline client/startup/i18n.js]: As a side effect of not invoking `setLanguage()`, the reactive `currentLanguage` variable is not changed and the form login will always show a suggested language. When nor user and admin language are set, e.g., the `English version` link is always present.

File: client/startup/i18n.js [hunk 3]
```diff
@@ -41,6 +41,9 @@ Meteor.startup(() => {
 	});
 
 	const applyLanguage = (language = 'en') => {
+		if (!language) {
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (sampaiodiego) [reply client/startup/i18n.js]: true.. during the development I got one situation where the param was `null` but with current implementation it is always set.. I'll remove this. thx

File: client/startup/i18n.js (hunk 2 above)
Comment (sampaiodiego) [reply client/startup/i18n.js]: actually `setLanguage` is still being called by the login form: https://github.com/RocketChat/Rocket.Chat/blob/8bfa5988068b7dae1a2ce16dca487a9dfa7a4a6d/packages/rocketchat-ui-login/client/login/footer.js#L33

this is the only place that still calls it.

**Comment by sampaiodiego** [comment]
@tassoevan pls review it again =)

**Pull request merged**